### ISSUE TITLE
Start transaction error after reconnected

### ISSUE THREAD

**Issue by Genuineh** [fsharplang-ru/pulsar-client-dotnet#285]
Pulsar Version : 3.3.2
Client Version : 3.6.0
Pulsar cluster run in k8s and client connect with pulsar-proxy

#  Problem
Start transaction by client will error after reconnect

# Step
1. send message in a loop with a new transaction
2. restart pulsar broker and run `./bin/pulsar zookeeper-shell delete /loadbalance/brokers/pulsar-broker-0.pulsar-broker.pulsar.svc.cluster.local:8080` in zookeeper pod
3. wait the pulsar broker pod restart success and observer

# Expected
reconnect and work normally

# Actual
1.client is connected
2.producer is connected
3.but error when starting transaction by client

# Code
``` csharp
using Microsoft.Extensions.Logging;
using Pulsar.Client.Api;

namespace PulsarProducerTest;

internal class ProducerTest
{
    static async Task Main(string[] args)
    {
        try
        {
            await OnlyCreateClient();
        }
        catch (Exception ex)
        {
            Console.WriteLine("Exception: " + ex.Message);
        }
    }

    internal static async Task OnlyCreateClient()
    {
        var loggerFactory = LoggerFactory.Create(builder =>
            builder.SetMinimumLevel(LogLevel.Debug).AddConsole()
        );
        PulsarClient.Logger = loggerFactory.CreateLogger("PulsarLogger");

        const string serviceUrl = "pulsar://192.168.1.116:30418";
        const string topicName = "golang-test-topic-2";
        var client = await new PulsarClientBuilder()
            .OperationTimeout(TimeSpan.FromSeconds(600))
            .ServiceUrl(serviceUrl)
            .EnableTransaction(true)
            .BuildAsync();
        var producer = await client.NewProducer(Schema.STRING())
            .SendTimeout(TimeSpan.Zero)
            .Topic(topicName)
            .CreateAsync();

        var statusThread = new Thread(() => PrintStatus(client, producer));
        statusThread.Start();

        while (true)
        {
            try
            {
                var transaction = await client.NewTransaction().BuildAsync();
                var message = producer.NewMessage("Sent from C# at " + DateTime.UtcNow.ToString(), txn: transaction);
                var messageId = await producer.SendAsync(message);
                Console.WriteLine("| ------------------- Send Msg ----------------- |");
                Console.WriteLine("| Message was sent with MessageId={0} |", messageId);
                Console.WriteLine("| ---------------------------------------------- |");
                await transaction.Commit();
            }
            catch (NotConnectedException ex)
            {
                Console.WriteLine("| ------------- NotConnectedException ------------- |");
                Console.WriteLine("| Exception:  |" + ex.Message);
                Console.WriteLine("| ------------------------------------------------ |");
            }
            catch (Exception ex)
            {
                Console.WriteLine("| ------------- Exception ------------- |");
                Console.WriteLine("| Exception:  |" + ex.Message);
                Console.WriteLine("| ExceptionType: |" + ex.GetType());
                Console.WriteLine("| StackTrace: |" + ex.StackTrace);
                Console.WriteLine("| ------------------------------------- |");
            }
            finally
            {
                await Task.Delay(3000);
            }
        }
    }

    static async void PrintStatus(PulsarClient client, IProducer<string> producer)
    {
        while (true)
        {
            Console.WriteLine("| ------------- Status ------------- |");
            Console.WriteLine("| Client is closed {0} |", client.IsClosed);
            var producerIsConnected = await producer.IsConnected();
            Console.WriteLine("| Producer is connected {0} |", producerIsConnected);
            Console.WriteLine("| ---------------------------------- |");
            await Task.Delay(3000);
        }
    }
}
```

# Log

## Try Reconnect
```log
warn: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) Socket was disconnected normally while reading
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) ChannelInactive, currently IsActive=True
info: PulsarLogger[0]
      Connection backgroundTask LogicalAddres Unspecified/192.168.1.230:31698 removed
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) fail requests
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) sendMb stopped
info: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) sendMb mailbox has stopped normally
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) has stopped
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) operationsMb stopped
dbug: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) readSocket stopped
info: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) requestsMb mailbox has stopped normally
info: PulsarLogger[0]
      clientCnx(7, LogicalAddres Unspecified/192.168.1.230:31698) operationsMb mailbox has stopped normally
| ------------- Exception ------------- |
Exception: tmsHandler-15 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected False          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-0 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected False          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-1 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected False          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-2 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected False          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-3 ConnectionHandler not connected
| ------------------------------------- |
dbug: PulsarLogger[0]
      producer(0, , -1) ConnectionHandler Starting reconnect to persistent://public/default/golang-test-topic-2
info: PulsarLogger[0]
      Connecting to Broker (LogicalAddres Unspecified/192.168.1.230:31698, PhysicalAddress Unspecified/192.168.1.230:31698) with maxMessageSize: 5242880
dbug: PulsarLogger[0]
      Socket connecting to Unspecified/192.168.1.230:31698
dbug: PulsarLogger[0]
      Socket connected to Unspecified/192.168.1.230:31698
dbug: PulsarLogger[0]
      Connection established for Unspecified/192.168.1.230:31698
dbug: PulsarLogger[0]
      clientCnx(8, LogicalAddres Unspecified/192.168.1.230:31698) Started read socket
dbug: PulsarLogger[0]
      clientCnx(8, LogicalAddres Unspecified/192.168.1.230:31698) Sending message of type Connect
info: PulsarLogger[0]
      Connected to Broker (LogicalAddres Unspecified/192.168.1.230:31698, PhysicalAddress Unspecified/192.168.1.230:31698)
dbug: PulsarLogger[0]
      clientCnx(8, LogicalAddres Unspecified/192.168.1.230:31698) Got message of type Connected
info: PulsarLogger[0]
      clientCnx(8, LogicalAddres Unspecified/192.168.1.230:31698) Connected ProtocolVersion: 19 ServerVersion: Pulsar Server3.3.2 MaxMessageSize: 0
```

## Reconnected But Start Transaction Error
```
| ------------- Exception ------------- |
Exception: tmsHandler-12 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected True          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-13 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected True          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-14 ConnectionHandler not connected
| ------------------------------------- |
| ------------- Status ------------- |
| Client is closed False               |
| Producer is connected True          |
| ---------------------------------- |
| ------------- Exception ------------- |
Exception: tmsHandler-15 ConnectionHandler not connected
| ------------------------------------- |
dbug: PulsarLogger[0]
      clientCnx(9, LogicalAddres Unspecified/192.168.1.230:31698) keepalive tick
dbug: PulsarLogger[0]
      clientCnx(9, LogicalAddres Unspecified/192.168.1.230:31698) Sending message of type Ping
dbug: PulsarLogger[0]
      clientCnx(9, LogicalAddres Unspecified/192.168.1.230:31698) Got message of type Pong
dbug: PulsarLogger[0]
      clientCnx(9, LogicalAddres Unspecified/192.168.1.230:31698) Got message of type Ping
dbug: PulsarLogger[0]
      clientCnx(9, LogicalAddres Unspecified/192.168.1.230:31698) Sending message of type Pong
```

## Total Log
[total.log](https://github.com/user-attachments/files/18055147/producer-6.log)

**Comment by Lanayx** [comment]
@Genuineh Hi, the header of the issue talks about the error in reconnect, how is the zookeeper command related to that? Are you trying to corrupt transaction state? Also, what is Java client behavior in the same situation? And also please check 3.6.1 client behavior, there was a reconnection fix in that version.

**Comment by Genuineh** [comment]
> @Genuineh Hi, the header of the issue talks about the error in reconnect, how is the zookeeper command related to that? Are you trying to corrupt transaction state? Also, what is Java client behavior in the same situation? And also please check 3.6.1 client behavior, there was a reconnection fix in that version.

@Lanayx broker restart successfully in k8s need to zookeeper data update (I think it maybe is a bug or our wrong), I think I didn't corrupt the transaction state I just try to create a scenario for recovery after the broker crash. In the same case, the go client work normally after broker recovery, java client maybe as the same as go client in this case. If you need, I can test java client in this case. And I will try the new version in this case. Thanks your reply.

**Comment by Genuineh** [comment]
@Lanayx This problem still exists in version 3.6.1.

**Comment by Lanayx** [comment]
@Genuineh I've just published a likely fix in https://www.nuget.org/packages/Pulsar.Client/3.6.2, can you please verify

**Comment by Genuineh** [comment]
Fixed in 3.6.2, thanks for your efforts, you solved the problem very quickly.
